File: app_frontend/src/types/reporte.ts
// src/types/reporte.ts

export interface ReporteFacturaItem {
    factura_id: number;
    numero_orden?: string | null;
    numero_caso?: string | null;
    fecha_emision: string; // Las fechas llegan como strings
  
    honorarios_generados: number;
    gastos_generados: number;
  
    vendedor_id: number;
    vendedor_nombre: string;
    vendedor_rut: string;
  
    cliente_id: number;
    cliente_razon_social: string;
    cliente_rut: string;

    // --- CAMPO AÑADIDO PARA EL BONO ---
    bono_calculado?: number; 
  }
  
export interface SumatoriaPorVendedor {
    vendedor_id: number;
    vendedor_nombre: string;
    total_honorarios: number;
}

// --- INTERFAZ CORREGIDA Y UNIFICADA ---
export interface ReporteResponse {
    items: ReporteFacturaItem[];
    total_count: number;
    sumatoria_total_honorarios: number;
    sumatorias_por_vendedor: SumatoriaPorVendedor[];
}

File: app_backend/app/schemas/reporte.py
# app/schemas/reporte.py
from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import date, datetime

# Schema para una única fila del reporte
class ReporteFacturaItem(BaseModel):
    factura_id: int
    numero_orden: Optional[str] = None
    numero_caso: Optional[str] = None
    fecha_emision: datetime
    honorarios_generados: float
    gastos_generados: float
    vendedor_id: int
    vendedor_nombre: str
    vendedor_rut: str
    cliente_id: int
    cliente_razon_social: str
    cliente_rut: str

    class Config:
        from_attributes = True

# --- NUEVOS SCHEMAS PARA SUMATORIAS ---
# Schema para la sumatoria por vendedor
class SumatoriaPorVendedor(BaseModel):
    vendedor_id: int
    vendedor_nombre: str
    total_honorarios: float

# Schema para la respuesta completa de la API, ahora con sumatorias
class ReporteResponse(BaseModel):
    items: List[ReporteFacturaItem]
    total_count: int
    sumatoria_total_honorarios: float # Suma total de honorarios en la respuesta
    sumatorias_por_vendedor: List[SumatoriaPorVendedor] # Lista de sumas por vendedor